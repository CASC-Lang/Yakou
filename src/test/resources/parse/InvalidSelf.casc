class InvalidSelf

impl InvalidSelf {
    new(self)
}