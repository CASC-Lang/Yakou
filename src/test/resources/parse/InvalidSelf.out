error: Constructor has already added `self` variable under the hood
--> ./InvalidSelf.casc:4:8
3 | impl InvalidSelf {
4 |     new(self)
            ^^^^= hint: Remove `self` keyword
5 |
error: `self` can only be at the start of parameters
--> ./InvalidSelf.casc:6:16
5 |
6 |     new(i: i32, self)
                    ^^^^= hint: move `self` to the start of parameters
7 |
error: `self` can only be at the start of parameters
--> ./InvalidSelf.casc:8:18
7 |
8 |     fn a1(i: i32, self) {}
                      ^^^^= hint: move `self` to the start of parameters